src/compat-transpiler/transforms/compat-unicode-property-transform.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * The MIT License (MIT)
+ * Copyright (c) 2019-present Junliang Huang <jlhwung@gmail.com>
+ */
+
+'use strict';
+
+/**
+ * A regexp-tree plugin to translate `/\p{ASCII_Hex_Digit}/u` to `/[0-9A-Fa-f]/`.
+ *
+ */
+
+function getRegenerateSets(node) {
+  return [
+    'regenerate-unicode-properties',
+    node.binary ? 'Binary_Property' : node.canonicalName,
+    node.binary ? node.canonicalName : node.canonicalValue,
+  ].join('/');
+}
+
+function negateCodepoints(codepoints) {
+  const newCodepoints = codepoints.slice();
+  if (newCodepoints[0] !== 0) {
+    newCodepoints.unshift(0);
+  } else {
+    newCodepoints.shift();
+  }
+  if (newCodepoints[newCodepoints.length - 1] !== 0x110000) {
+    newCodepoints.push(0x110000);
+  } else {
+    newCodepoints.pop();
+  }
+  return newCodepoints;
+}
+
+function printCodepoint(codepoint) {
+  if (codepoint < 65536) {
```

Comment:
More readable: `<= 0xFFFF`

(in general, use hexadecimal integer literals for code points)